data/kernels/denoiseprofile.cl

```diff
@@ -86,16 +86,23 @@ denoiseprofile_dist(read_only image2d_t in, global float* U4, const int width, c
   const int y = get_global_id(1);
   const int gidx = mad24(y, width, x);
 
-  // Reminder: q.x and q.y can be negative
   if(x >= width || y >= height) return;
-  if(x+q.x >= width || y+q.y >= height) return;
-  if(x+q.x < 0 || y+q.y < 0) return;
+
+  int xpq = x + q.x;
+  int ypq = y + q.y;
+  // Convert out of bounds indexes to 0
+  // Reminder: q.x and q.y can be negative
+  xpq *= (x+q.x < width && x+q.x >= 0) ? 1.0f : 0.0f;
```

Comment:
I suggest to use int constants here.

Reply:
Oops, indeed.
I perfectly agree, thanks!